rudder-agent/SOURCES/rudder.init

```diff
@@ -0,0 +1,241 @@
+#!/bin/sh
+
+#==============================================
+# Start/stop script for all Rudder components #
+#==============================================
+#
+### BEGIN INIT INFO
+# Provides:		rudder
+# Required-Start:	$local_fs $remote_fs $network $syslog
+# Required-Stop:	$local_fs $remote_fs $network $syslog
+# Default-Start:	2 3 4 5
+# Default-Stop:		0 1 6
+# Short-Description:	Rudder server
+# Description:		Rudder server including all components required to run
+### END INIT INFO
+#
+# Copyright (C) 2011-2015 Normation
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# GPL License: http://www.gnu.org/licenses/gpl.txt
+#
+#====================================================================
+
+# Utility functions
+
+get_service_name()
+{
+# Arguments:
+## Service name
+  if type systemctl >/dev/null 2>&1; then
+    SERVICE_NAME=$(systemctl -a | awk -F'.' '{print $1}' | grep "${1}" | tail -n 1)
+  elif type chkconfig >/dev/null 2>&1; then
+    SERVICE_NAME=$(chkconfig 2>/dev/null | awk '{ print $1 }' | grep "${1}" | tail -n 1)
+  else
+    SERVICE_NAME=$(ls -1 /etc/init.d | grep "${1}" | tail -n 1)
+  fi
+
+  echo ${SERVICE_NAME}
+}
+
+service_action()
+{
+# Arguments:
+## Service name
+## Action to trigger (start/stop/...)
+## Is the role present on this machine (0/1)
+
+# If the role is present, run the action
+if [ "${3}" -eq 1 ]; then
+  printf " * ${1}...\t"
+  service "${1}" "${2}" >/dev/null 2>&1 || RETURN="${?}"
```

Comment:
Here RETURN is a global variable without initial value.
So it will always be empty or the content of the last error, so if there is an error on a service, the next service will be said to have an error too